src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -456,4 +505,28 @@ private function hasShippingMethodWithFee(ShippingMethodInterface $shippingMetho
 
         return false;
     }
+
+    private function getPossiblePaymentMethods($paymentMethodName): array
+    {
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findCartByTokenValue($this->sharedStorage->get('cart_token'));
+
+        if(!$order->getLastPayment()) {
+            return $this->paymentMethodRepository->findByName($paymentMethodName, 'US');
```

Comment:
Why have you added this logic? If an order does not have any payment, there are no payment methods available. Also, it returns totally different content of the array, so we cannot be sure about its content. Either it should be moved to separate method or (preferably) this if statements should be removed

Reply:
Sorry some dead code I have left :) still this 'if' is needed because if there is no payments it should return empty array.